### ISSUE TITLE
DNS Filtering not working when macOS is connected to the hotspot of iOS

### ISSUE THREAD

**Issue by TriStaRvOiD** [AdguardTeam/AdguardForMac#1357]
### AdGuard version

2.12.2.1415 release

### OS version

macOS Sonoma 14.2

### Issue Details

Steps to reproduce:
1. Turn on Personal Hotspot in iOS
2. Connect macOS to the hotspot
3. Turn on Adguard for Mac (with DNS Filtering on)

### Expected Behavior

Internet/DNS FIltering works.

### Actual Behavior

Internet/DNS FIltering does not work.

### Screenshots

<details><summary>Screenshot 1:</summary>

<img width="1001" alt="Screenshot 2023-12-14 at 5 47 56 PM" src="https://github.com/AdguardTeam/AdguardForMac/assets/58824912/5e35b2bc-294f-4745-95e7-6987c3e353b1">

</details>

<details><summary>Screenshot 2:</summary>

<img width="1039" alt="Screenshot 2023-12-14 at 6 21 03 PM" src="https://github.com/AdguardTeam/AdguardForMac/assets/58824912/bbe21f20-7cde-4d36-a026-51fcd9c48205">

</details>

### Additional Information

* iOS doesn't have Adguard
* I am using iOS 17.2
* For some reason this problem seems to have started only after I updated to 14.2 from 14.1
* It's possible that DNS Filtering has never actually worked on the hotspot. But now the internet itself has stopped working altogether.
* I have Lockdown Mode turned on, do tell me if I should test this again after disabling.

**Comment by TriStaRvOiD** [comment]
Similar to #1073, but I don't see what this has to do with Adguard on iOS.

**Comment by AlexandrPkhm** [comment]
Hi @TriStaRvOiD 

Unfortunately, we are unable to reproduce the problem. Could you please try to reproduce the issue with Lockdown mode turned off, as it may interfere with connectivity and therefore hotspot mode? Please let us know if the issue persists with lockdown mode enabled.

**Comment by TriStaRvOiD** [comment]
Hello.

It's not working for me even after disabling Lockdown mode on both devices.

**Comment by AlexandrPkhm** [comment]
@TriStaRvOiD 

In this particular case, we need to get the app logs to diagnose this issue. Here's what we need you to do:

1. Click **AdGuard icon** in the menu bar → Gear → Advanced  → Logging → Logging level → **Debug**;
2. Reproduce the issue and remember the exact time it happened;
3. Menu → Advanced → Logging → **Export Logs and System Info…**;
4. Send the archive to `apple@adguard.com` and mention this issue number in the subject.

We also would also like to receive an export of your application settings. To get it please click **AdGuard icon** in the menu bar → Gear → Advanced  → Export settings… and sent the generated file to us along with the logs archive.

**Comment by TriStaRvOiD** [comment]
I've sent the mail, regarding the issue number though - I mentioned it in the body.

**Comment by AlexandrPkhm** [comment]
@TriStaRvOiD 

Could you please check if the issue still persists on the latest version of AdGuard for Mac, which is version 2.13? It is available via this [link](https://github.com/AdguardTeam/AdguardForMac/releases/tag/v2.13.0). Please let us know of the results.

**Comment by TriStaRvOiD** [comment]
I updated the app through the built-in updater and the issue persisted. So then I uninstalled the app, re-downloaded the .dmg and did a clean install (skipped the initial onboarding set up by choosing "Leave as is"). The issue persists.

Something is really wrong because even setting the DNS protection to default is causing no internet.
<img width="656" alt="Screenshot 2023-12-25 at 10 17 31 PM" src="https://github.com/AdguardTeam/AdguardForMac/assets/58824912/3966ed3f-1617-4d97-ad61-6d564dbdee2d">
